pkg/tlscheck/tlscheck.go

```diff
@@ -18,9 +19,24 @@ const (
 	timeout = 2 * time.Second
 )
 
+// addrValid validates the URL. It assumes scheme prefixes have been removed
+func addrValid(addr string) error {
+	addrNoScheme := urlfmt.TrimHTTPPrefixes(addr)
+	// url.Parse requires scheme to trigger the correct variant of parsing (it has two)
+	_, err := url.Parse("https://" + addrNoScheme)
+	return err
+}
+
 // CheckTLS checks if the address is using TLS
 func CheckTLS(ctx context.Context, origAddr string) (bool, error) {
-	addr := urlfmt.TrimHTTPPrefixes(origAddr)
+	addr := strings.TrimSpace(origAddr)
+	addr = urlfmt.TrimHTTPPrefixes(addr)
+	// Ellimitate obvious mistakes in the host name
+	addr = strings.TrimSpace(addr)
```

Comment:
Regarding using `TrimSpace` here, this will force `CheckTLS` to succeed for urls such as `" example.com"`, however other usages of that URL after the check are likely to fail.

For example, thinking of the local/delegated scanning scenario, if `CheckTLS` succeeds then the invalid URL is upserted into Sensor's registry store.  Later that invalid URL will be used for pulling metadata/layers from the image registry (potentially running into similar issues that spawned this change).  If `CheckTLS` were the fail instead, then the upsert would never happen and later usages prevented.

Does it makes sense to remove `TrimSpace` from `CheckTLS` and instead rely on an error? or perhaps move the `TrimSpace` to higher up in the call flow so all 'usages' can benefit from the sanitized url? (for the pull secret use case, that would be when the secrets are ingested).

```suggestion
	addr = urlfmt.TrimHTTPPrefixes(addr)
```

Reply:
Oh right, I totally missed that. I had the space removal in `func (s *secretDispatcher) processDockerConfigEvent` but then I changed it. Let me readd it there, so that the clean URL is inserted into the store.

Reply:
Please check [a762d4e](https://github.com/stackrox/stackrox/pull/11668/commits/a762d4ee9d2475762729c45e2a4c8b51ce3b0bc8) (and the current diff)